sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/SymbolicExecution/CSharpExplodedGraph.cs

```diff
@@ -341,7 +344,7 @@ protected override void VisitInstruction(ExplodedGraphNode node)
                 case SyntaxKind.StackAllocArrayCreationExpression:
                     {
                         var sv = new SymbolicValue();
-                        newProgramState = sv.SetConstraint(ObjectConstraint.NotNull, newProgramState);
+                        newProgramState = newProgramState.SetConstraint(sv, ObjectConstraint.NotNull);

```

Comment:
You could probably merge the different calls instead of having always `variable = variable.Foo()`.

Reply:
I didn't wanted to refactor the whole code base, because to be consistent everywhere I would have to chain all calls to ProgramState in all classes. I would prefer to do this incrementally, not in a couple of huge commits

Reply:
Ok